pkg/nfd/nodefeaturerule.go

```diff
@@ -0,0 +1,250 @@
+package nfd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	nfdv1 "github.com/openshift/node-feature-discovery/api/nfd/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NodeFeatureRuleBuilder provides a struct for NodeFeatureRule object
+// from the cluster and a NodeFeatureRule definition.
+type NodeFeatureRuleBuilder struct {
+	// Builder definition. Used to create
+	// Builder object with minimum set of required elements.
+	Definition *nfdv1.NodeFeatureRule
+	// Created Builder object on the cluster.
+	Object *nfdv1.NodeFeatureRule
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before Builder object is created.
+	errorMsg string
+}
+
+// NewNodeFeatureRuleBuilderFromObjectString creates a Builder object from CSV alm-examples.
+func NewNodeFeatureRuleBuilderFromObjectString(apiClient *clients.Settings, almExample string) *NodeFeatureRuleBuilder {
+	glog.V(100).Infof(
+		"Initializing new Builder structure from almExample string")
+
+	if apiClient == nil {
+		glog.V(100).Info("The apiClient of the Policy is nil")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(nfdv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Info("Failed to add nfd v1 scheme to client schemes")
+
+		return nil
+	}
+
+	nodeFeatureRule, err := getNodeFeatureRuleFromAlmExample(almExample)
+
+	glog.V(100).Infof(
+		"Initializing Builder definition to NodeFeatureRule object")
+
+	nodeFeatureRuleBuilder := NodeFeatureRuleBuilder{
+		apiClient:  apiClient,
+		Definition: nodeFeatureRule,
+	}
+
+	if err != nil {
+		glog.V(100).Infof(
+			"Error initializing NodeFeatureRule from alm-examples: %s", err.Error())
+
+		nodeFeatureRuleBuilder.errorMsg = fmt.Sprintf("error initializing NodeFeatureRule from alm-examples: %s",
+			err.Error())
+
+		return &nodeFeatureRuleBuilder
+	}
+
+	if nodeFeatureRuleBuilder.Definition == nil {
+		glog.V(100).Infof("The NodeFeatureRule object definition is nil")
+
+		nodeFeatureRuleBuilder.errorMsg = "NodeFeatureRule definition is nil"
+
+		return &nodeFeatureRuleBuilder
+	}
+
+	return &nodeFeatureRuleBuilder
+}
+
+// FeatureRulePull loads an existing NodeFeatureRuleBuilder into Builder struct.
+func FeatureRulePull(apiClient *clients.Settings, name, namespace string) (*NodeFeatureRuleBuilder, error) {
```

Comment:
```suggestion
func PullFeatureRule(apiClient *clients.Settings, name, namespace string) (*NodeFeatureRuleBuilder, error) {
```
consistency with other packages